js/filterMovies.js

```diff
@@ -0,0 +1,136 @@
+import { getMovies, getShowTimes } from "./myAPI.js";
+
+export async function filterMovies(parameter) {
+	const movieTimes = await getShowTimes();
+	const movies = await getMovies();
+
+	const newMovieArray = movies.map((movie, index) => {
+		return {
+			...movie,
+			showtime: []
+		};
+	});
+
+	movieTimes.forEach((salon) => {
+		const { name: salonName } = salon;
+		salon.dates.forEach((date) => {
+			let dateString = date.date.slice(0, 10);
+
+			date.showtimes.forEach((date) => {
+				let obj = {
+					time: new Date(`${dateString} ${date.time}`),
+					salon: salonName
+				};
+				newMovieArray[date.movieIndex].showtime.push(obj);
+			});
+		});
+	});
+
+	let results = [];
+	let dates, now, num, movieRelease, tomorrow;
+
+	switch (parameter) {
+		case "visningstider":
+			now = new Date();
+			now.setHours(0, 0, 0, 0);
+
+			tomorrow = new Date();
+			tomorrow.setDate(now.getDate() + 1);
+			tomorrow.setHours(23, 59, 59, 59);
+
+			movieTimes.forEach((salon) => {
+				salon.dates.forEach((date) => {
+					const showtime = new Date(date.date).getTime();
+					if (now.getTime() > showtime && showtime < tomorrow.getTime()) {
```

Comment:
It should be the other way arround  showtime > now.getTime()  otherwise it pass dates that are before now :D

Reply:
👀 Noooooooo

Fixed